test/spec/services/file-loader.spec.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+describe('Service: fileLoader', function () {
+
+  // load the service's module
+  beforeEach(function () {
+    module('angularWidgetInternal');
+    module({
+      tagAppender: jasmine.createSpy('tagAppender').andCallFake(function (filename) {
+        return filename + '.done';
+      }),
+      sequentialFileLoader: {
+        loadFiles: jasmine.createSpy().andCallFake(function (arr) {
+         return arr.length;
+       })
+      }
+    });
+  });
+
+  // instantiate service
+  var fileLoader, $rootScope, $q;
+  beforeEach(inject(function (_fileLoader_, _$rootScope_, _$q_) {
+    fileLoader = _fileLoader_;
+    $rootScope = _$rootScope_;
+    $q = _$q_;
+  }));
+
+  describe('when processing parralel script loading (non-array parameters)', function () {
+      it('should handle files and arrays correctly', inject(function (tagAppender, sequentialFileLoader) {
```

Comment:
where are you testing that when all promises are resolved, the returned promise is resolved?